src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -162,12 +162,10 @@ public void ValidateAppInsightsTelemetryConnectionString(RuntimeConfig runtimeCo
         ValidateConfigProperties();
         ValidatePermissionsInConfig(runtimeConfig);
 
-        // If the ConfigValidationExceptions list doesn't contain a DataApiBuilderException with connection string error message,
-        // then only we run the metadata validation.
-        if (!ConfigValidationExceptions.Any(x => x.Message.Equals(DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE)))
-        {
-            await ValidateEntitiesMetadata(runtimeConfig, loggerFactory);
```

Comment:
why are you no longer calling `ValidateEntitiesMetadata`? aren't there additional validations there that aren't covered by your entity relationship validation?

Reply:
I am not sure if its because the intention of that function changed at some point, but this is what it looks like before my changes.

![image](https://github.com/Azure/data-api-builder/assets/93220300/6faa9da8-b6fb-4071-9583-dd74754cb933)


As you can see it really is just validating the entity relationships with the ValidateRelationshipInConfig call.

So to more easily understand, and I will add this to the PR description, what is going on here, you can think of `ValidateEntitiesMetadata` as being replaced by the `ValidateEntityRelationshipsInConfigWithGeneratedMetadata` which has an internal call to the replacement for `ValidateRelationshipsInConfig` which is now the `ValidateEntityRelationshipsInConfigWithProvidedMetadata`, and then we also add separately to these functions `ValidateEntityRelationshipsInConfigWithoutDBMetadata` which runs on its own to do the validation that doesnt need a metadataprovider. So in the case that we have DB errors trying to connect, we only run the function that has no need for a metadata provider, and then in the other cases we still call the same equivalent functions they just have different names now that more closely describe their behavior and responsibilities.

Reply:
`await metadataProviderFactory.InitializeAsync()` initializes a metadataprovider, and the metadataprovider initialization executes validations. The other validations are abstracted away by the Init() but they are happening.

ValidationRelationshipsInConfig is just another validation that occurs.